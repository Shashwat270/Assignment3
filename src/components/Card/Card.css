body, html {
    height: 100%;
    width: 100%;
    font-weight: 10;
    margin: 0px;
  }
.card {
    max-width: 300px;
    /* flex-basis: 300px;
    flex-grow: 0; */
    align-self: flex-start;
    text-align: center;
    font-family: arial;
    /* border : 2px solid gray; */
    margin-top: 2%;
    margin-bottom: 5%;
    margin-left: auto;
    margin-right: auto;
  }
  .book-image {
    width : 300px;
    height: 220px;
}
  .book-name {
    font-size: 20px;
    font-weight: 500;
    color: black;
    padding-left: 10%;
    padding-right: 10%;
    margin-bottom: 8px;
    text-align: justify;
}
.rating-data {
    float:left;
    padding: 3px;
    color: black;
    font-weight: 500;
    font-size: 1.5rem;
}
.heart-logo {
    float: right;
    padding: 3px 10px 10px 5px;
    width: 8%;
}
