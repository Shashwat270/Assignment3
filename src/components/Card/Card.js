import React, { Component } from 'react';
import './Card.css';
import PropTypes from 'prop-types';

const blackHeart = require('./../../Assets/heart-black.svg');
const redHeart = require('./../../Assets/heart-red.svg');
const imagePath = require('./../../Assets/cover_image.jpeg');

class Card extends Component {
  render() {
    const bookName = this.props.data.Name;
    const ratingsData = this.props.data.rating;
    const likedData = true;

    return (
      <div className="card">
        <img src={imagePath} alt="Book" className="book-image" />
        <p className="book-name">{bookName}</p>

        <span className="rating-data">{ratingsData}</span>
        <img src={(likedData === true) ? redHeart : blackHeart} alt="Heart" className="heart-logo" />
      </div>
    );
  }
}
Card.defaultProps = {
  Name: 'Harry Potter',
  rating: 3.9,
};

export default Card;
