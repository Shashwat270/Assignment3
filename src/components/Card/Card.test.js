import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('should render one exactly card', () => {
    const tree = renderer.create(<Card />).toJSON();
    expect(tree).toMatchSnapshot();
  });
//  const wrapper = shallow(<Card />);
//   it('should not have any props to be empty', () => {
//     // console.log(wrapper.instance().props.data);
//     const propData = wrapper.instance().props.data;
//     expect(propData).toBeDefined();
//   });
});
