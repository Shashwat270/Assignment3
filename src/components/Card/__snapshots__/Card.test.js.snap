// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render one exactly card 1`] = `
<div
  className="card"
>
  <img
    alt="Book"
    className="book-image"
    src="cover_image.jpeg"
  />
  <p
    className="book-name"
  >
    Harry Potter and the sorcerer stone
  </p>
  <span
    className="rating-data"
  >
    4.5
  </span>
  <img
    alt="Heart"
    className="heart-logo"
    src={
      Object {
        "ReactComponent": [Function],
        "__esModule": true,
        "default": "heart-red.svg",
      }
    }
  />
</div>
`;
